presentation/src/main/java/com/smile/presentation/MainViewModel.kt

```diff
@@ -20,19 +19,24 @@ class MainViewModel @Inject constructor(
     private val imageUseCase: UpdateLocalImageUseCase
 ) : BaseViewModel() {
 
-    private val _characterLiveData = MutableLiveData<Wrapper<List<CharacterUI>>>()
-    val characterLiveData: LiveData<Wrapper<List<CharacterUI>>> = _characterLiveData
+    private val _characterLiveData = MutableLiveData<List<CharacterUI>>()
+    val characterLiveData: LiveData<List<CharacterUI>> = _characterLiveData
+    private val liveDataHandler = LiveDataHandler(failure)
 
     init {
         getCharacters()
     }
 
     private fun getCharacters() {
         scope.launch(handler()) {
-            val resp = getCharactersUseCase(Unit).run {
-                CharacterToUIMapper().mapObject(this)
+            val resp = getCharactersUseCase()
+            uiScope.launch {
+                liveDataHandler.parse(resp) { data ->
+                    data?.let {
```

Comment:
What happens in the UI if data is null?

Reply:
@chizoba no case for now... most probably should have some sort of empty state... what I can do is emit an error for now sha